packages/jest-types/src/Circus.ts

```diff
@@ -16,7 +16,9 @@ export type BlockMode = void | 'skip' | 'only' | 'todo';
 export type TestMode = BlockMode;
 export type TestName = Global.TestName;
 export type TestFn = Global.TestFn;
-export type HookFn = (done?: DoneFn) => Promise<any> | null | undefined;
+export type HookFn = (
```

Comment:
wWhy the asymmetry? `Promise<unknown>` and `null` instead of `Promise<any>` like in `Global.TestFn`, `Global.HookFn`, and `Circus.TestFn`

Reply:
this was mostly me fixing a type error since it didn't accept `void`. Good point on the mismatch though - I'll push up exporting `HookFn` from `Globals` and reuse that here.

The type should ideally be `Promise<void> | void`, but then we don't get type errors if you return anything else. From my testing, `void` only matters to the caller, the callee doesn't have to fulfill that contract, it's just that the returned value cannot be used

Reply:
Hmm, looks like it does matter https://www.typescriptlang.org/play?#code/MYewdgzgLgBAHgLhgNxASwCYwLwwCwBMA3AFABmArmMFGuDGQBQCUSqmMA3iTLzAE4BTKBX5h8xEgF8gA

Reply:
Oh, nice! Can change then 👍 Should probably keep `Promise<unknown>` just so people can do `test('t', () => promiseThing())`

Reply:
Actually, nevermind, this might be the problem const f: () => void = () => 42;

Reply:
ah, there you go! Adding `|undefined` makes it behave like we want, though